File: Examples/DotNet/CSharp/Worksheet/SortWorksheetData.cs
using System;
using Com.Aspose.Cells.Api;
using Com.Aspose.Cells.Model;
using Com.Aspose.Storage.Api;
using System.Collections.Generic;

namespace Worksheet
{
    class SortWorksheetData
    {
        public static void Run()
        {
            // ExStart:1
            CellsApi cellsApi = new CellsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "Sample_Test_Book.xls";
            String sheetName = "Sheet1";
            String cellArea = "A5:A10";
            String storage = "";
            String folder = "";

            SortKey sort = new SortKey();
            sort.Key = 0;
            sort.SortOrder = "descending";

            DataSorter body = new DataSorter();
            body.CaseSensitive = "false";
            body.HasHeaders = "false";
            body.SortLeftToRight = "false";
            body.KeyList = new List<SortKey> { sort };

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));

                // Invoke Aspose.Cells Cloud SDK API to sort worksheet data
                SaaSposeResponse apiResponse = cellsApi.PostWorksheetRangeSort(fileName, sheetName, cellArea, storage, folder, body);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("Worksheet data is sorted!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }

            // ExEnd:1
        }
    }
}

File: Examples/Python/SortWorkSheetData.py
import asposecellscloud
from asposecellscloud.CellsApi import CellsApi
from asposecellscloud.CellsApi import ApiException
from asposecellscloud.models import SortKey
from asposecellscloud.models import DataSorter

import asposestoragecloud
from asposestoragecloud.StorageApi import StorageApi

apiKey = "XXXXX" #sepcify App Key
appSid = "XXXXX" #sepcify App SID
apiServer = "http://api.aspose.com/v1.1"
data_folder = "../../data/"

#Instantiate Aspose Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, True)
storageApi = StorageApi(storage_apiClient)
#Instantiate Aspose Cells API SDK
api_client = asposecellscloud.ApiClient.ApiClient(apiKey, appSid, True)
cellsApi = CellsApi(api_client);

#set input file name
filename = "Sample_Test_Book.xls"
sheetName = "Sheet1"
cellArea = "A5:A10"

sort = SortKey.SortKey()
sort.Key = 0
sort.SortOrder = "descending"

body = DataSorter.DataSorter()
body.CaseSensitive = "false"
body.HasHeaders = "false"
body.SortLeftToRight = "false"
body.KeyList = [sort]

#upload file to aspose cloud storage
storageApi.PutCreate(Path=filename, file=data_folder + filename)

try:
    #invoke Aspose.Cells Cloud SDK API to sort worksheet data
    response = cellsApi.PostWorksheetRangeSort(name=filename, sheetName=sheetName, cellArea=cellArea, body=body)

    if response.Status == "OK":
        #download Workbook from storage server
        response = storageApi.GetDownload(Path=filename)
        outfilename = "c:/temp/" +  "sorted_" + filename
        with open(outfilename, 'wb') as f:
                    for chunk in response.InputStream:
                        f.write(chunk)

except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message
